### PR TITLE
General repo improvements

### PR DESCRIPTION
**Pull request by vzotova**
* fancier event check
* solhint rules + hook
* prettier action + hook

### CODE REVIEW

File: .prettierignore
Comment (theref) [inline .prettierignore:1]: How come we want to ignore these files?

File: .solhintignore
Comment (theref) [inline .solhintignore:1]: Again, why these files? (not an issue, genuinely curious) :)

File: tests/test_subscription_manager.py [hunk 1]
```diff
@@ -103,16 +104,17 @@ def test_create_policy_with_sponsor(subscription_manager, accounts, chain):
...
-    assert len(events) == 1
-    event = events[0]
-    assert bytes(event["policyId"]) == policy_id
-    assert event["sponsor"] == sponsor
-    assert event["owner"] == alice
-    assert event["size"] == size
-    assert event["startTimestamp"] == start
-    assert event["endTimestamp"] == end
-    assert event["cost"] == fee
+    assert tx.events == [
```
Comment (theref) [inline tests/test_subscription_manager.py:107]: love this way of checking events 🚀

**Review by theref — APPROVED** [review]

File: .solhintignore
Comment (vzotova) [reply .solhintignore:1]: Same answer as above

File: contracts/contracts/lib/ReEncryptionValidator.sol [hunk 2]
```diff
@@ -72,64 +73,76 @@ library ReEncryptionValidator {
...
-            _precomputed.pointEZxCoord,     // zE_x
-            _precomputed.pointEZyCoord),    // zE_y
+        require(
+            ecmulVerify(
+                _capsule.pointE.xCoord, // E_x
+                _precomputed.pointEyCoord, // E_y
+                _cFrag.proof.bnSig, // z
+                _precomputed.pointEZxCoord, // zE_x
+                _precomputed.pointEZyCoord
+            ), // zE_y
```
Comment (derekpierre) [inline contracts/contracts/lib/ReEncryptionValidator.sol:97]: Minor suggestion where I disagree with the linter.

```suggestion
                _precomputed.pointEZyCoord // zE_y
            ),
```

Some similar instances in later lines in this file.

File: contracts/test/AdjudicatorTestSet.sol [hunk 3]
```diff
@@ -2,27 +2,29 @@
 
 pragma solidity ^0.8.0;
 
-
 import "../contracts/Adjudicator.sol";
 import "../contracts/lib/SignatureVerifier.sol";
-//import "../contracts/proxy/Upgradeable.sol";
 
+//import "../contracts/proxy/Upgradeable.sol";
```
Comment (derekpierre) [inline contracts/test/AdjudicatorTestSet.sol:8]: Perhaps we just remove this line?

File: contracts/xchain/PolygonChild.sol [hunk 4]
```diff
@@ -21,7 +17,9 @@ contract PolygonChild is FxBaseChildTunnel, Ownable {
         bytes memory data
     ) internal override validateSender(sender) {
         emit MessageReceived(sender, data);
-        (bool success, /* returnId */ ) = stakeInfoAddress.call(data);
+        //(bool success, /* returnId */ ) =
+        // solhint-disable-next-line avoid-low-level-calls
+        stakeInfoAddress.call(data);
```
Comment (derekpierre) [inline contracts/xchain/PolygonChild.sol:22]: Can we now remove L20?

File: tests/application/test_authorization.py [hunk 5]
```diff
@@ -63,11 +63,11 @@ def test_authorization_increase(accounts, threshold_staking, taco_application, s
 
     # Check that all events are emitted
     events = taco_application.AuthorizationIncreased.from_receipt(tx)
-    assert len(events) == 1
-    event = events[0]
-    assert event["stakingProvider"] == staking_provider
-    assert event["fromAmount"] == 0
-    assert event["toAmount"] == value
+    assert events == [
```
Comment (derekpierre) [inline tests/application/test_authorization.py:66]: So much cleaner - 🚀

**Review by derekpierre — APPROVED** [review]
🎸

File: contracts/contracts/lib/ReEncryptionValidator.sol (hunk 2 above)
Comment (vzotova) [reply contracts/contracts/lib/ReEncryptionValidator.sol:97]: Looks like there is open issue in prettier for that https://github.com/prettier/prettier/issues/807
We can add hack for ignoring this line for prettier or just accept that. Wdyt?

File: contracts/xchain/PolygonChild.sol (hunk 4 above)
Comment (vzotova) [reply contracts/xchain/PolygonChild.sol]: ✔️

File: contracts/test/AdjudicatorTestSet.sol (hunk 3 above)
Comment (vzotova) [reply contracts/test/AdjudicatorTestSet.sol]: ✔️

File: contracts/contracts/lib/UmbralDeserializer.sol [hunk 6]
```diff
@@ -60,24 +58,24 @@ library UmbralDeserializer {
...
-    uint256 constant POINT_SIZE = 33;
-    uint256 constant SIGNATURE_SIZE = 64;
-    uint256 constant CAPSULE_SIZE = 2 * POINT_SIZE + BIGNUM_SIZE;
-    uint256 constant CORRECTNESS_PROOF_SIZE = 4 * POINT_SIZE + BIGNUM_SIZE + SIGNATURE_SIZE;
-    uint256 constant CAPSULE_FRAG_SIZE = 3 * POINT_SIZE + BIGNUM_SIZE;
-    uint256 constant FULL_CAPSULE_FRAG_SIZE = CAPSULE_FRAG_SIZE + CORRECTNESS_PROOF_SIZE;
-    uint256 constant PRECOMPUTED_DATA_SIZE = (20 * BIGNUM_SIZE) + 32 + 20 + 5;
+    uint256 internal constant BIGNUM_SIZE = 32;
+    uint256 internal constant POINT_SIZE = 33;
+    uint256 internal constant SIGNATURE_SIZE = 64;
```
Comment (jMyles) [inline contracts/contracts/lib/UmbralDeserializer.sol:66]: Yeah?  Not DER encoded?

File: tests/application/test_authorization.py [hunk 7]
```diff
@@ -210,11 +210,11 @@ def test_involuntary_authorization_decrease(
     assert stake_info.stakes(staking_provider)[STAKE_INFO_OPERATOR_SLOT] == ZERO_ADDRESS
 
     events = taco_application.AuthorizationInvoluntaryDecreased.from_receipt(tx)
-    assert len(events) == 1
-    event = events[0]
-    assert event["stakingProvider"] == staking_provider
-    assert event["fromAmount"] == value // 2
-    assert event["toAmount"] == authorization
+    assert events == [
+        taco_application.AuthorizationInvoluntaryDecreased(
```
Comment (jMyles) [inline tests/application/test_authorization.py:214]: badass.  Love seeing events represented this way.

**Review by jMyles — APPROVED** [review]
As always, some of the nitty-gritty of solidity is a little tough for me to gork quickly, but I can verify that there's nothing here that looks crazy.  :-)

**Pull request merged**